src/view/popup/usePopup.ts

```diff
@@ -1,28 +1,27 @@
-import { useEffect, useState } from "react";
+import { useContext, useEffect, useState } from "react";
 import {
     CalculationData,
     ICalculationsRepository,
     UserType,
 } from "../../data/calculations/ICalculationsRepository";
-import { CountryName } from "../../data/constants/CountryEmissions";
-import { ISelectedCountriesRepository } from "../../data/selected-countries/ISelectedCountriesRepository";
-import { useMountEffect } from "./useOnceAfterFirstMount";
+import { SelectedCountriesContext } from "../provider/RootProvider";
 import { backgroundStopRecordingBytes } from "./utils/backgroundStopRecordingBytes";
 import { calculateAverageSpecificEmissionsHelper } from "./utils/calculateAverageSpecificEmissions";
 import { calculateCarbon } from "./utils/calculateCarbon";
 import { refreshActiveTabAndRecordBytes } from "./utils/refreshActiveTabAndRecordBytes";
 
 export const usePopup = () => {
-    const selectedCountriesRepository: ISelectedCountriesRepository =
-        ISelectedCountriesRepository.instance;
+    const selectedCountriesContext = useContext(SelectedCountriesContext);
+    if (selectedCountriesContext === null) {
+        throw Error("SelectedCountriesContext is null");
+    }
```

Comment:
You can probably add this inside root provider by creating a hook called `useRootProvider` and doing this check first before `return selectedCountriesContext`